src/main/java/apaintus/controllers/FigureLogController.java

```diff
@@ -0,0 +1,101 @@
+package apaintus.controllers;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+import javafx.util.Callback;
+
+import java.util.List;
+
+import apaintus.models.ListViewCell;
+import apaintus.models.commands.Invoker;
+import apaintus.models.commands.SelectCommand;
+import apaintus.models.shapes.DrawableShape;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+
+public class FigureLogController implements ChildController<Controller> {
+	private Controller controller;
+	private CanvasController canvasController;
+	private Invoker invoker;
+
+	public static final ObservableList<String> observableListFigures = FXCollections.observableArrayList();
+
+	@FXML
+	private ListView<String> figureList;
+	private List<DrawableShape> shapeList;
+
+	private static final int UP = -1;
```

Comment:
Variable ordering in a class:
1. static finals
2. fxml
3. project specific objects
4. rest

Reply:
Thank you for the reminder, didn't pay attention to the order